sdk/storage/azure-storage-files-datalake/src/datalake_utilities.cpp

```diff
@@ -39,7 +39,10 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake { nam
     std::string result;
     for (const auto& pair : dataLakePropertiesMap)
     {
-      result.append(pair.first + "=" + Internal::Base64EncodeText(pair.second) + ",");
+      result.append(
+          pair.first + "="
+          + Azure::Core::Base64Encode(std::vector<uint8_t>(pair.second.begin(), pair.second.end()))
```

Comment:
Does `pair.second` need to be a `std::string`?

From the issue:
> For example, does this `Storage::Metadata` map represent a string key with a binary data value? If so, it should be changed to `std::string, vector<uint8_t>`:
https://github.com/Azure/azure-sdk-for-cpp/blob/b212d287abafcc4e9cf06d78ac57dd87286227c7/sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp#L68

Reply:
Yes, in most of the cases our users will be using `std::string` for metadata. Plus it aligns with all other SDK

Reply:
> Yes, in most of the cases our users will be using `std::string` for metadata. Plus it aligns with all other SDK

I am not familiar with this concept too much, can you share how it aligns with other SDK?

Is the `Metadata` here in our SDK identical to the concept in C#? I ask because I don't see them encoding string to Base64 string. So either their dictionary already contains the base64 encoded string, in which case we may want to do something similar, or change the map to `vector<uint8_t>`

https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-properties-metadata?tabs=dotnet#set-and-retrieve-metadata
https://raw.githubusercontent.com/Azure/azure-sdk-for-net/master/sdk/storage/Azure.Storage.Blobs/src/Generated/BlobRestClient.cs
```C#
if (metadata != null) {
    foreach (System.Collections.Generic.KeyValuePair<string, string> _pair in metadata)
    {
        _request.Headers.SetValue("x-ms-meta-" + _pair.Key, _pair.Value);
    }
}
```

Reply:
There're two ways to send metadata over the wire. One used by blob service is just like in the .Net code you shared. The other one used by DataLake is putting all metadata in a single header. See https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/filesystem/setproperties#request-headers